core/account-id/src/lib.rs

```diff
@@ -1,6 +1,69 @@
+//! This crate provides a type for representing a valid, unique account identifier on the [NEAR](https://near.org) network.
+//!
+//! ## Account ID Rules
+//!
+//! - Minimum length is `2`
+//! - Maximum length is `64`
+//! - An **Account ID** consists of **Account ID parts** separated by `.`, example:
+//!   - `root` ✔
+//!   - `alice.near` ✔
+//!   - `app.stage.testnet` ✔
+//! - Must not start or end with separators (`_`, `-` or `.`):
+//!   - `_alice.` ✗
+//!   - `.bob.near-` ✗
+//! - Each part of the **Account ID** consists of lowercase alphanumeric symbols separated either by `_` or `-`, example:
+//!   - `ƒelicia.near` ✗ (`ƒ` is not `f`)
+//!   - `1_4m_n0t-al1c3.near` ✔
+//! - Separators are not permitted to immediately follow each other, example:
+//!   - `alice..near` ✗
+//!   - `not-_alice.near` ✗
+//! - An **Account ID** that is 64 characters long and consists of lowercase hex characters is a specific **implicit account ID**
+//!
+//! Learn more here: <https://docs.near.org/docs/concepts/account#account-id-rules>
```

Comment:
Same here, I'd probably just add a link to the reference here, rather than duplicate the reference.